mlir/test/Dialect/Bufferization/Transforms/one-shot-bufferize-analysis.mlir

```diff
@@ -172,7 +172,7 @@ func.func @materialize_in_destination_aliasing(%t: tensor<?xf32>, %p1: index, %p
   %dest = tensor.extract_slice %t[%p2][5][1] : tensor<?xf32> to tensor<5xf32>
   // CHECK: bufferization.materialize_in_destination
   // CHECK-SAME: {__inplace_operands_attr__ = ["true", "true"]}
-  %r = bufferization.materialize_in_destination %src in %dest : tensor<5xf32>
+  %r = bufferization.materialize_in_destination %src in %dest : (tensor<5xf32>, tensor<5xf32>) -> tensor<5xf32>
```

Comment:
Not sure this actually reads better.
Using only the type of the tensor of the memref is fully unambiguous, no?

Reply:
This is because the result is optional. In case of a `memref`, there is no result (like `linalg.generic`). TableGen helpers such as `AllTypesMatch` etc. do not work with that. I could write the printer/parser in C++ though, then we could stay with the original syntax.